strawberry/schema/schema.py

```diff
@@ -163,6 +164,7 @@ def __init__(
         self._schema._strawberry_schema = self  # type: ignore
 
         self._warn_for_federation_directives()
+        self._validate_types_node_id_attr()
```

Comment:
shall we call this `_resolve_node_ids`?

Reply:
also, do you think we can merge this with the code at line 128?

<img width="709" alt="CleanShot 2023-06-12 at 20 50 17@2x" src="https://github.com/strawberry-graphql/strawberry/assets/667029/9c460153-eeea-4e92-8f18-2e90749ba079">

Reply:
> shall we call this `_resolve_node_ids`?

Sure! I didn't like the name I gave it, that is better :)

> also, do you think we can merge this with the code at line 128?

I initially tried to do that, but the iterated `types` in there are just extra types that you can pass to the schema and not all existing types :(